test/extensions/wasm/wasm_test.cc

```diff
@@ -73,6 +73,52 @@ TEST(WasmTest, BadSignature) {
                             "Bad function signature for: _proxy_onConfigure");
 }
 
+TEST(WasmTest, Segv) {
+  Stats::IsolatedStoreImpl stats_store;
+  Api::ApiPtr api = Api::createApiForTest(stats_store);
+  Upstream::MockClusterManager cluster_manager;
+  Event::SimulatedTimeSystem time_system;
+  Event::DispatcherImpl dispatcher(time_system, *api);
+  auto wasm = std::make_shared<Extensions::Common::Wasm::Wasm>("envoy.wasm.vm.wavm", "", "",
+                                                               cluster_manager, dispatcher);
+  EXPECT_NE(wasm, nullptr);
+  const auto code = TestEnvironment::readFileToStringForTest(
+      TestEnvironment::substitute("{{ test_rundir }}/test/extensions/wasm/test_data/segv.wasm"));
+  EXPECT_FALSE(code.empty());
+  auto context = std::make_unique<TestContext>(wasm.get());
+  EXPECT_CALL(*context, scriptLog(spdlog::level::err, Eq("before badptr")));
+  EXPECT_TRUE(wasm->initialize(code, "<test>", false));
+  wasm->setGeneralContext(std::move(context));
+  EXPECT_THROW_WITH_MESSAGE(wasm->start(), Extensions::Common::Wasm::WasmException,
+                            "emscripten llvm_trap");
+}
+
+TEST(WasmTest, DivByZero) {
+  Stats::IsolatedStoreImpl stats_store;
+  Api::ApiPtr api = Api::createApiForTest(stats_store);
+  Upstream::MockClusterManager cluster_manager;
+  Event::SimulatedTimeSystem time_system;
+  Event::DispatcherImpl dispatcher(time_system, *api);
+  auto wasm = std::make_shared<Extensions::Common::Wasm::Wasm>("envoy.wasm.vm.wavm", "", "",
+                                                               cluster_manager, dispatcher);
+  EXPECT_NE(wasm, nullptr);
+  const auto code = TestEnvironment::readFileToStringForTest(
+      TestEnvironment::substitute("{{ test_rundir }}/test/extensions/wasm/test_data/segv.wasm"));
+  EXPECT_FALSE(code.empty());
+  auto context = std::make_unique<TestContext>(wasm.get());
+  EXPECT_CALL(*context, scriptLog(spdlog::level::err, Eq("before div by zero")));
+  EXPECT_TRUE(wasm->initialize(code, "<test>", false));
+  wasm->setGeneralContext(std::move(context));
+  wasm->wasmVm()->start(wasm->generalContext());
+#if 0
+  // Not yet supported in gtes.
+  EXPECT_THROW_WITH_MESSAGE_THAT(wasm->generalContext()->onLog(), Extensions::Common::Wasm::WasmException,
```

Comment:
Nit: `EXPECT_THROW_WITH_MESSAGE` not `EXPECT_THROW_WITH_MESSAGE_THAT`.

Reply:
The problem here is that `EXPECT_THROW_WITH_MESSAGE` doesn't support matchers, and I need to do a string prefix.  EXPECT_THROW_WITH_MESSAGE_THAT is in a proposed PR for gtest/gmock.

I compromised and I just check that we were able to translate the WAVM Exception into an envoy WasmException.

Reply:
Wouldn't `EXPECT_THROW_WITH_REGEX()` work? i.e.
```
EXPECT_THROW_WITH_REGEX(wasm->generalContext()->onLog(), Extensions::Common::Wasm::WasmException,
                        "wavm.integerDivideByZeroOrOverflow.*");
```

Reply:
The reason I need the string prefix is that the stack trace is in the remained of the message.

Reply:
Yeah, hence the `".*"` (regex) to catch reminder of the output.

Reply:
FWIW, I've added the snippet from my previous comment to the test and it passes.

Reply:
Didn't see you message.  Yes that works, Done.